saleor/payment/gateways/authorize_net/plugin.py

```diff
@@ -0,0 +1,188 @@
+            "help_text": "Determines if Saleor should automatically capture payments.",
+            "label": "Automatic payment capture",
+        },
+        "supported_currencies": {
+            "type": ConfigurationTypeField.STRING,
+            "help_text": "Determines currencies supported by gateway."
+            " Please enter currency codes separated by a comma.",
+            "label": "Supported currencies",
+        },
+    }
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        configuration = {item["name"]: item["value"] for item in self.configuration}
+        self.config = GatewayConfig(
+            gateway_name=GATEWAY_NAME,
+            auto_capture=configuration["automatic_payment_capture"],
+            supported_currencies=configuration["supported_currencies"],
+            connection_params={
+                "api_login_id": configuration["api_login_id"],
+                "transaction_key": configuration["transaction_key"],
+                "client_key": configuration["client_key"],
+                "use_sandbox": configuration["use_sandbox"],
+            },
+            store_customer=configuration["store_customers_card"],
+        )
+
+    def _get_gateway_config(self) -> GatewayConfig:
+        return self.config
+
+    @classmethod
+    def validate_plugin_configuration(cls, plugin_configuration: "PluginConfiguration"):
+        """Validate if provided configuration is correct."""
+        configuration = {
+            item["name"]: item["value"] for item in plugin_configuration.configuration
+        }
+
+        api_login_id = configuration.get("api_login_id", None)
+        transaction_key = configuration.get("transaction_key", None)
+
+        # Only check when both are set, otherwise the dashboard is hard to use
+        if api_login_id and transaction_key:
+            success, message = authenticate_test(
+                api_login_id, transaction_key, configuration.get("use_sandbox")
+            )
+            if not success:
+                raise ValidationError(
+                    message, code=PluginErrorCode.PLUGIN_MISCONFIGURED.value,
+                )
+
+    @require_active_plugin
+    def authorize_payment(
+        self, payment_information: "PaymentData", previous_value
+    ) -> "GatewayResponse":
+        return authorize(payment_information, self._get_gateway_config())
+
+    @require_active_plugin
+    def capture_payment(
+        self, payment_information: "PaymentData", previous_value
+    ) -> "GatewayResponse":
+        return capture(payment_information, self._get_gateway_config())
+
+    @require_active_plugin
+    def refund_payment(
+        self, payment_information: "PaymentData", previous_value
+    ) -> "GatewayResponse":
+        try:
+            payment = Payment.objects.get(pk=payment_information.payment_id)
+        except Payment.DoesNotExist:
+            raise PaymentError("Cannot find Payment.")
```

Comment:
I would add here the ID of the payment which cannot be found.
`f"Cannot find Payment {payment_information.payment_id}.`